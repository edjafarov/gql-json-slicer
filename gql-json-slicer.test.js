const gqlSlice = require("./gql-json-slicer");

describe('gqlSlice', () => {
  var data1 = {
    field1: "field1",
    field2: "field2"
  }
  test('filters out field', () => {
    expect(gqlSlice(`{field1}`, data1)).toMatchSnapshot();
  })

  test('ignore fields not specified', () => {
    expect(gqlSlice(`{
      field1
      field3
    }`, data1)).toMatchSnapshot();
  })
  var data2 = {
    field1: "field1",
    field2: ["field2"]
  }
  test('arrays should be handled properly', () => {
    expect(gqlSlice(`{field2}`, data2)).toMatchSnapshot();
  })
  var data3 = {
    field1: "field1",
    field2: {
      innerField1: "innerField1",
      innerField2: "innerField2",
      deeperField: {
        deeperInner1: "deeperInner1",
        deeperInner2: "deeperInner2"
      }
    }
  }

  test('deep nested fields should be resolved', () => {
    expect(gqlSlice(`{
      field2
    }`, data3)).toMatchSnapshot();
  })

  test('deep nested fields should be deeply resolved', () => {
    expect(gqlSlice(`{
      field2 {
        innerField2
        deeperField {
          deeperInner1
        }
      }
    }`, data3)).toMatchSnapshot();
  })

  var data4 = {
    field1: "field1",
    field2: [{
      innerField1: "innerField11",
      innerField2: "innerField12",
    },{
      innerField1: "innerField21",
      innerField2: "innerField22",
    },{
      innerField1: "innerField31",
      innerField2: "innerField32",
    }]
  }

  test('deep nested fields should be resolved', () => {
    expect(gqlSlice(`{
      field2 {
        innerField2
      }
    }`, data4)).toMatchSnapshot();
  })

})
