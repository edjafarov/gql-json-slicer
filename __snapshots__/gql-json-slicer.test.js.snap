// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gqlSlice arrays should be handled properly 1`] = `
Object {
  "field2": Array [
    "field2",
  ],
}
`;

exports[`gqlSlice deep nested fields should be deeply resolved 1`] = `
Object {
  "field2": Object {
    "deeperField": Object {
      "deeperInner1": "deeperInner1",
    },
    "innerField2": "innerField2",
  },
}
`;

exports[`gqlSlice deep nested fields should be resolved 1`] = `
Object {
  "field2": Object {
    "deeperField": Object {
      "deeperInner1": "deeperInner1",
      "deeperInner2": "deeperInner2",
    },
    "innerField1": "innerField1",
    "innerField2": "innerField2",
  },
}
`;

exports[`gqlSlice deep nested fields should be resolved 2`] = `
Object {
  "field2": Array [
    Object {
      "innerField2": "innerField12",
    },
    Object {
      "innerField2": "innerField22",
    },
    Object {
      "innerField2": "innerField32",
    },
  ],
}
`;

exports[`gqlSlice filters out field 1`] = `
Object {
  "field1": "field1",
}
`;

exports[`gqlSlice ignore fields not specified 1`] = `
Object {
  "field1": "field1",
}
`;
